src/components/class-menu-container.tsx

```diff
@@ -38,53 +41,49 @@ export class ClassMenuContainer extends BaseComponent <IProps, {}> {
     const {user} = this.stores;
     const classNames = uniq(user.portalClassOfferings.map(o => o.className));
     const currentProblemOrdinal = this.getCurrentProblemOrdinal();
-    const links: IMenuLink[] = [];
+    const links: IDropdownItem[] = [];
 
     // If, by chance, there are no classes in the offerings, return a link
     // with a name matching the current class name and no link. This is bit of
     // code is primarily for when the application is running in demo mode.
     if (classNames.length === 0) {
-      return {
-        currentTitle:  user.className,
-        links: [
-          {
-          title: user.className
-          }
-        ]
-      };
+      return [];
     }
 
     // For each class (that is, for each item in the class switcher menu) we
     // try to find a problem/offering with the same ordinal identifier as the
     // current problem. If we find a match, we add that one to the link array.
     // If not, we use the first one in the list of offerings for that class.
-
     classNames.forEach( (className) => {
       const classLinks = user.portalClassOfferings.filter(o => o.className === className);
       const matchingLink = classLinks.find( l => l.problemOrdinal === currentProblemOrdinal);
+      const handleClick = (name: string, url: string) => {
+        window.location.replace(url);
+        const log = {
+          event: LogEventName.DASHBOARD_SWITCH_CLASS,
+          parameters: {className, link: url}
+        };
+        Logger.log(log.event, log.parameters, LogEventMethod.DO);
```

Comment:
Doh.  

I hadn't thought too hard about this, because realistically this log method isn't going to go through at all.  In the past we had passed a callback to navigate away, but the log handling code didn't return ... logging had ~15s latency in some cases so it was reported as a [bug](https://www.pivotaltracker.com/story/show/169345486).

Doug had spiked into a few possible solutions, but ultimately he went with this (only in the correct order, as you point out):

https://github.com/concord-consortium/collaborative-learning/commit/2905cbc042710385cde594c20f0075db500fd5cb

In any case, this is a good reminder that we probably want to reload content without reloading the App at some point. The gotcha with that is possibly having to get a different token from the Portal.